.travis.yml

```diff
@@ -8,20 +8,66 @@ ghc:
   - "8.0"
   - "7.10"
 
-before_deploy: |
-  if [[ $TRAVIS_HASKELL_VERSION == "8.6.5" ]]; then
-    cabal install -f 'static'
-    tar -C dist/build/dotenv -cvf dotenv-linux-x86_64-static.tar.gz dotenv
-  fi
-
-deploy:
-  provider: releases
-  api_key: "$GITHUB_TRAVIS_DEPLOY"
-  file: "dotenv-linux-x86_64-static.tar.gz"
-  skip_cleanup: true
-  release_notes_file: "CHANGELOG.md"
-  edge: true
-  draft: true
-  on:
-    condition: $TRAVIS_HASKELL_VERSION == "8.6.5"
-    tags: true
+env:
+  global:
+    - PATH=$HOME/.local/bin:$PATH
+    # Envs for OS X
+    - GHC_VER="8.6.5"
+    - CABAL_VER="2.4.0.0"
+    - PATH=$HOME/Library/Python/2.7/bin:$PATH
+    - PATH=$HOME/.cabal/bin:$HOME/.ghcup/bin:$PATH
+    - PATH=$HOME/.ghcup/ghc/$GHC_VER:$HOME/.cabal/$CABAL_VER:$PATH
+
+before_install:
+  - pip install --user awscli
+  - echo $(which pip)
+  - mkdir -p ~/$TRAVIS_BUILD_NUMBER
+  - aws s3 sync s3://dotenv-releases/$TRAVIS_BUILD_NUMBER ~/$TRAVIS_BUILD_NUMBER
+
+jobs:
+  include:
+    - stage: OS X build and test
+      language: generic
+      os: osx
+      addons: skip
+      before_install: sh .travis/ghcup.sh
+      install: cabal install --only-dependencies --enable-tests
+      script: cabal configure --enable-tests && cabal build && cabal test
+
+    - stage: OS X binary build
+      if: branch = master
```

Comment:
Is it possible to replace this comparison with this https://docs.travis-ci.com/user/customizing-the-build#building-specific-branches

Reply:
I think that filters the branches for any build @juanpaucar 
The syntax `if: branch = name` is the only option we have to set a condition for jobs and stages right now: https://docs.travis-ci.com/user/build-stages/#specifying-stage-order-and-conditions